esmvaltool/utils/cmorizers/obs/cmorize_obs_ESACCI-OC.py

```diff
@@ -0,0 +1,192 @@
+# pylint: disable=invalid-name
+"""ESMValTool CMORizer for ESACCI-OC data.
+
+Tier
+
+Source
+   ftp://oceancolour.org/occci-v3.1/geographic/netcdf/monthly/chlor_a/
+
+Last access
+   20190227
+
+Download and processing instructions
+   Put all files under a single directory (no subdirectories with years)
+   in ${RAWOBS}/Tier2/ESACCI-OC
+
+Modification history
+   20190227-A_lova_to: written.
+
+"""
+
+import logging
+import os
+import glob
+import xarray as xr
+
+import iris
+import numpy as np
+
+from .utilities import (_set_global_atts, _fix_coords, _fix_var_metadata,
+                        _read_cmor_config, _save_variable, constant_metadata)
+
+logger = logging.getLogger(__name__)
+
+# read in CMOR configuration
+CFG = _read_cmor_config('ESACCI-OC.yml')
+
+
+def _fix_data(cube, var):
+    """Specific data fixes for different variables."""
+    logger.info("Fixing data ...")
+    with constant_metadata(cube):
+        if var == 'chl':
+            cube *= 1.e-06
+    return cube
+
+
+def _fix_attr_cmip5(out_dir, var):
+    """Adjust for CMIP5 standard"""
+    logger.info("Fix to CMIP5 standard...")
+    in_file = glob.glob(out_dir + '/OBS*' + var + '*.nc')[0]
```

Comment:
it'd be more Pythonic and faster (albeit here it's only a few files) if you used `os.walk` instead eg:
```
import fnmatch
import os

matches = []
for root, dirnames, filenames in os.walk(out_dir):
    for filename in fnmatch.filter(filenames, 'OBS*' + var + '*.nc'):
        matches.append(os.path.join(root, filename))
```

Reply:
@valeriupredoi I was doing this lazy search since I needed the final output filename to modify attributes... but it really doesn't make a lot of sense.
In the commit 4142eec, I added to `_save_variable` function in `utilities.py` the return of the output filename that is in turn passed to the `_fix_attr_cmip5` thus avoiding this file search.
How does it looks to you now?

Reply:
this part of the code was removed .